### ISSUE TITLE
Generic executable/configuration scripts for KDCluster 

### ISSUE THREAD

**Issue by kmathur2** [bluek8s/kubedirector#495]
Need a mechanism to execute scripts for running pods at the role level, this can be really handy for application developers to install/configure/reconfigure packages.

**Comment by joel-bluedata** [comment]
Are you thinking of ways to do things that would normally be done by "kubectl exec"? If so, is this to make it easier to do in some way, or to work around exec-privileges issues?

**Comment by kmathur2** [comment]
Yes, this is to avoid series of identical kubectl execs for a kdcluster with a large number of pods requiring package install, etc. Also for bookkeeping/logging of what was done inside pods post deployments. 

But you raised a good point if the kubectl exec is blocked for the namespace should we allow this? The intent is definitely not to bypass the RBACs but more of a convenience feature for kdclusters. We should explore if there are ways to implement this without breaking access control.
